### PR TITLE
Set CMAKE_OSX_DEPLOYMENT_TARGET on mac builds

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by sbc100** [comment]
I don't really know how to test this...

**Comment by dschuff** [comment]
I'll try it locally. You can tell with `otool` what the target is.

**Comment by dschuff** [comment]
```
Load command 9
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.12
      sdk 10.13
```

File: src/build.py [hunk 1]
```diff
@@ -814,11 +817,6 @@ def LLVM():
       '-DLLVM_ENABLE_TERMINFO=%d' % (not IsLinux()),
   ])
 
-  if IsMac():
```
Comment (dschuff) [inline src/build.py]: I guess we also need to remove the one that you added to fastcomp too?

**Comment by sbc100** [comment]
Presumably this also removes the futimes dependency?

**Comment by dschuff** [comment]
I would assume so. I'd guess that the headers you get when you set 10.12 as the deployment target should match what the patform actually has.

**Pull request merged**